src/averageAges.js

```diff
@@ -55,7 +66,31 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const getPeople = onlyWithSon
+    ? people
+      .filter(person => person.sex === 'm')
+    : people;
+
+  let len = 0;
+
+  const sum = getPeople.reduce((ttl, person) => {
+    const mother = people.find(m => m.name === person.mother);
+
+    if (mother) {
+      len++;
+
+      return ttl + person.born - mother.born;
+    }
+
+    return ttl;
+  }, 0);
+
+  return sum / len;
+}
+
+function averageAge(people) {
+  return people.reduce((sum, person) => sum + person.died - person.born, 0)
```

Comment:
person object can be de-structured in callback args
```suggestion
  return people.reduce((sum, {died, born}) => sum + died - born, 0)
```